chatkit-core/src/main/kotlin/com/pusher/chatkit/model/mappers/Room.kt

```diff
@@ -0,0 +1,35 @@
+package com.pusher.chatkit.model.mappers
+
+import com.pusher.chatkit.model.network.ReadStateApiType
+import com.pusher.chatkit.model.network.RoomApiType
+import com.pusher.chatkit.model.network.RoomMembershipApiType
+import com.pusher.chatkit.rooms.Room
+
+internal fun mapToRoom(
+        room: RoomApiType,
+        memberships: RoomMembershipApiType?,
+        readState: ReadStateApiType?
+) = mapToRoom(
+        room = room,
+        memberships = memberships?.userIds.orEmpty().toSet(),
+        unreadCount = readState?.unreadCount
+)
+
+internal fun mapToRoom(
+        room: RoomApiType,
+        memberships: Set<String>?,
+        unreadCount: Int?
+) = Room(
+        id = room.id,
+        name = room.name,
+        createdById = room.createdById,
+        pushNotificationTitleOverride = room.pushNotificationTitleOverride,
+        isPrivate = room.private,
+        customData = room.customData,
+        createdAt = room.createdAt,
+        updatedAt = room.updatedAt,
+        deletedAt = room.deletedAt,
+        lastMessageAt = room.lastMessageAt,
+        unreadCount = unreadCount,
+        memberUserIds = memberships.orEmpty()
```

Comment:
using named params just introduces noise when it's clear what each param is in a given context (at a given call site)

Reply:
I find it safer/easier to refactor, especially when there is a run of `String`s for example (the various dates), but I don't feel particularly strongly about it.

Reply:
The IDE has "Change Signature" refactor option and will do it for you.